### PR TITLE
Fixed #28391 -- Fixed Cast() with CharField and max_length on MySQL.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28391

### CODE REVIEW

File: tests/db_functions/test_cast.py [hunk 1]
```diff
@@ -19,6 +22,12 @@ def test_cast_from_field(self):
         numbers = Author.objects.annotate(cast_string=Cast('age', models.CharField(max_length=255)),)
         self.assertEqual(numbers.get().cast_string, '1')
 
+    @mock.patch('sys.stderr')
+    @unittest.skipIf(connection.vendor == 'sqlite', "SQLite doesn't support Cast with precision.")
```
Comment (felixxm) [inline tests/db_functions/test_cast.py]: See https://www.sqlite.org/lang_expr.html#castexpr.

File: tests/db_functions/test_cast.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
-from django.db import models
+import unittest
```
Comment (atombrella) [inline tests/db_functions/test_cast.py]: `from unittest import mock, skipIf` instead?

File: tests/db_functions/test_cast.py [hunk 3]
```diff
@@ -19,6 +21,12 @@ def test_cast_from_field(self):
         numbers = Author.objects.annotate(cast_string=Cast('age', models.CharField(max_length=255)),)
         self.assertEqual(numbers.get().cast_string, '1')
 
+    @mock.patch('sys.stderr')
+    @skipIf(connection.vendor == 'sqlite', "SQLite doesn't support Cast with precision.")
```
Comment (felixxm) [inline tests/db_functions/test_cast.py]: See https://www.sqlite.org/lang_expr.html#castexpr.

Comment (timgraham) [inline tests/db_functions/test_cast.py]: I guess it could be better as a database feature in case any third-party databases also have that limitation.

**Comment by felixxm** [comment]
@timgraham Thanks for the review. I updated PR.

**Comment by felixxm** [comment]
buildbot

File: tests/db_functions/test_cast.py [hunk 4]
```diff
@@ -19,6 +19,13 @@ def test_cast_from_field(self):
         numbers = Author.objects.annotate(cast_string=Cast('age', models.CharField(max_length=255)),)
         self.assertEqual(numbers.get().cast_string, '1')
 
+    # MySQL backend might issue a warning.
```
Comment (timgraham) [inline tests/db_functions/test_cast.py]: Clarify "might"? Or simply say `# Silence "Truncated incorrect CHAR(1) value: 'Bob'".`

**Pull request merged**